litellm/llms/notdiamond.py

```diff
@@ -0,0 +1,253 @@
+    "openai/gpt-4": "gpt-4",
+    "openai/gpt-4-0613": "gpt-4-0613",
+    "openai/gpt-4o": "gpt-4o",
+    "openai/gpt-4o-2024-05-13": "gpt-4o-2024-05-13",
+    "openai/gpt-4-turbo": "gpt-4-turbo",
+    "openai/gpt-4-turbo-2024-04-09": "gpt-4-turbo-2024-04-09",
+    "openai/gpt-4-turbo-preview": "gpt-4-turbo-preview",
+    "openai/gpt-4-0125-preview": "gpt-4-0125-preview",
+    "openai/gpt-4-1106-preview": "gpt-4-1106-preview",
+    # anthropic
+    "anthropic/claude-2.1": "claude-2.1",
+    "anthropic/claude-3-opus-20240229": "claude-3-opus-20240229",
+    "anthropic/claude-3-sonnet-20240229": "claude-3-sonnet-20240229",
+    "anthropic/claude-3-5-sonnet-20240620": "claude-3-5-sonnet-20240620",
+    "anthropic/claude-3-haiku-20240307": "claude-3-haiku-20240307",
+    # mistral
+    "mistral/mistral-large-latest": "mistral/mistral-large-latest",
+    "mistral/mistral-medium-latest": "mistral/mistral-medium-latest",
+    "mistral/mistral-small-latest": "mistral/mistral-small-latest",
+    "mistral/codestral-latest": "mistral/codestral-latest",
+    "mistral/open-mistral-7b": "mistral/open-mistral-7b",
+    "mistral/open-mixtral-8x7b": "mistral/open-mixtral-8x7b",
+    "mistral/open-mixtral-8x22b": "mistral/open-mixtral-8x22b",
+    # perplexity
+    "perplexity/llama-3-sonar-large-32k-online": "perplexity/llama-3-sonar-large-32k-online",
+    # cohere
+    "cohere/command-r": "cohere_chat/command-r",
+    "cohere/command-r-plus": "cohere_chat/command-r-plus",
+    # google
+    "google/gemini-pro": "gemini/gemini-pro",
+    "google/gemini-1.5-pro-latest": "gemini/gemini-1.5-pro-latest",
+    "google/gemini-1.5-flash-latest": "gemini/gemini-1.5-flash-latest",
+    "google/gemini-1.0-pro-latest": "gemini/gemini-pro",
+    # replicate
+    "replicate/mistral-7b-instruct-v0.2": "replicate/mistralai/mistral-7b-instruct-v0.2",
+    "replicate/mixtral-8x7b-instruct-v0.1": "replicate/mistralai/mixtral-8x7b-instruct-v0.1",
+    "replicate/meta-llama-3-70b-instruct": "replicate/meta/meta-llama-3-70b-instruct",
+    "replicate/meta-llama-3-8b-instruct": "replicate/meta/meta-llama-3-8b-instruct",
+    # togetherai
+    "togetherai/Mistral-7B-Instruct-v0.2": "together_ai/mistralai/Mistral-7B-Instruct-v0.2",
+    "togetherai/Mixtral-8x7B-Instruct-v0.1": "together_ai/mistralai/Mixtral-8x7B-Instruct-v0.1",
+    "togetherai/Mixtral-8x22B-Instruct-v0.1": "together_ai/mistralai/Mixtral-8x22B-Instruct-v0.1",
+    "togetherai/Phind-CodeLlama-34B-v2": "together_ai/Phind/Phind-CodeLlama-34B-v2",
+    "togetherai/Llama-3-70b-chat-hf": "together_ai/meta-llama/Llama-3-70b-chat-hf",
+    "togetherai/Llama-3-8b-chat-hf": "together_ai/meta-llama/Llama-3-8b-chat-hf",
+    "togetherai/Qwen2-72B-Instruct": "together_ai/Qwen/Qwen2-72B-Instruct",
+}
+
+
+class NotDiamondError(Exception):
+    def __init__(self, status_code, message):
+        self.status_code = status_code
+        self.message = message
+        self.request = httpx.Request(
```

Comment:
You're setting `self.request` and `self.response` here as unvarying attributes. Do you know whether they're referenced elsewhere?

If they're included here for debugging, it might be best to accept them as constructor args, then pass them along from the failed request / response instead.

Reply:
I just followed their code style from other modules such as [cohere](https://github.com/BerriAI/litellm/blob/944f22a0891aec8a22c6fd5e005e65e3e915ce3b/litellm/llms/cohere.py#L12) or [gemini](https://github.com/BerriAI/litellm/blob/944f22a0891aec8a22c6fd5e005e65e3e915ce3b/litellm/llms/gemini.py#L22). I don't think they are being referenced elsewhere either